app/src/main/java/com/example/journeyjournal/Activities/CommentActivity.java

```diff
@@ -119,12 +136,57 @@ public void done(List<Comment> comments, ParseException e) {
                 for (Comment comment : comments) {
                     Log.i(TAG, "Comment: " + comment.getComment() + ", username: " + comment.getCommenter().getUsername());
                 }
-
                 // save received comments to list and notify adapter of change
                 allComments.clear();
                 allComments.addAll(comments);
                 adapter.notifyDataSetChanged();
             }
         });
     }
+
+    protected void querySavedComments() {
+        ParseQuery<Comment> query = ParseQuery.getQuery(Comment.class);
+        query.include(Post.KEY_USER);
+        query.addDescendingOrder(Post.KEY_CREATED_AT);
+        query.setSkip(0);
+        query.fromLocalDatastore();
+        query.findInBackground(new FindCallback<Comment>() {
+            @SuppressLint("NotifyDataSetChanged")
+            @Override
+            public void done(List<Comment> comments, ParseException e) {
+                Log.i(TAG, comments.toString());
+
+                // Remove the previously cached results.
+                Comment.unpinAllInBackground("Comments", new DeleteCallback() {
+                    public void done(ParseException e) {
+                        // Cache the new results.
+                        Comment.pinAllInBackground("Comments", comments);
+                    }
+                });
+
+                if (e != null) {
+                    Log.e(TAG, "Issue getting posts.", e);
+                    return;
+                }
+
+                // at this point, we have gotten the posts successfully
+                for (Comment comment : comments) {
+                    Log.i(TAG, "Comment: " + comment.getComment() + ", username: " + comment.getCommenter().getUsername());
+                }
+
+                allComments.clear();
+                allComments.addAll(comments);
+                adapter.notifyDataSetChanged();
+            }
+        });
+    }
+
+    private void whichQuery() {
```

Comment:
I think you can change this to be a bit more descriptive. maybe queryComments here and change the network call to be queryNetworkComments